modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/ScalaStepFilter.scala

```diff
@@ -11,11 +11,39 @@ import com.sun.jdi.Method
 import com.sun.jdi.ReferenceType
 
 import scala.jdk.CollectionConverters.*
+import java.util.Optional
 
 abstract class ScalaStepFilter(scalaVersion: ScalaVersion) extends StepFilter {
   protected def skipScalaMethod(method: Method): Boolean
+  def format(method: Method): Option[String] = {
+    if (method.isBridge) None
+    else if (isDynamicClass(method.declaringType)) None
+    else if (isJava(method)) Some(formatJava(method))
+    else if (isConstructor(method)) Some(formatJava(method))
+    else if (isStaticConstructor(method)) Some(formatJava(method))
+    else if (isAdaptedMethod(method)) None
+    else if (isAnonFunction(method)) Some(formatJava(method))
+    else if (isLiftedMethod(method)) Some(formatJava(method))
+    else if (isAnonClass(method.declaringType)) Some(formatJava(method))
+    // TODO in Scala 3 we should be able to find the symbol of a local class using TASTy Query
+    else if (isLocalClass(method.declaringType)) Some(formatJava(method))
+    else if (scalaVersion.isScala2 && isNestedClass(method.declaringType)) Some(formatJava(method))
+    else if (isDefaultValue(method)) Some(formatJava(method))
+    else if (isTraitInitializer(method)) Some(formatJava(method))
+    else formatScala(method)
+  }
+  def formatScala(method: Method): Option[String] = Some(formatJava(method))
+  def formatJava(method: Method): String = {
+    val declaringType = method.declaringType().name.split("\\.").last
+    val methodName = method.name()
+    val argumentTypes = method.argumentTypes.asScala.toList
+      .map(t => t.name().split("\\.").last)
+      .mkString(",")
+    s"$declaringType.$methodName(${if (argumentTypes.nonEmpty) argumentTypes else ""})"
```

Comment:
Maybe add the return type to be consistent with Scala methods:

```suggestion
    s"$declaringType.$methodName(${if (argumentTypes.nonEmpty) argumentTypes else ""}): $returnType"
```